embulk-core/src/main/java/org/embulk/spi/util/Reporters.java

```diff
@@ -0,0 +1,78 @@
+package org.embulk.spi.util;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.embulk.spi.Reporter;
+import org.embulk.spi.ReporterCloseable;
+
+import java.util.Map;
+
+public final class Reporters
```

Comment:
Any reasons to use the specific class `Reporters`, not like `Map<Reporter.Type, Reporter>` directly? (e.g. We use `List<FilterPlugin>` for a set (list) of filter plugins.)

I also have the same concern with `ReporterPlugins`. They may look open for plugin developers as the methods are `public` in `org.embulk.spi`. IIUC, they're actually not. 

If we really want this class, we may want to move this out of `org.embulk.spi`, and give a different name.